marimbabot_planning/scripts/self_calibration_optimization_example.ipynb

```diff
@@ -0,0 +1,355 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "id": "a00ad405",
+   "metadata": {},
+   "source": [
+    "# Marimba Model Fitting / Calibration"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "f142791f",
+   "metadata": {},
+   "source": [
+    "This notebook contains some example code to fit a marimba model onto a set of random hits with audio feedback.\n",
+    "\n",
+    "We optimize the spatial transformation matrix based on previously collected data."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 479,
+   "id": "720959d0",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import math\n",
+    "import numpy as np\n",
+    "from scipy.optimize import least_squares\n",
+    "\n",
+    "import matplotlib.pyplot as plt\n",
+    "%matplotlib inline"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 480,
+   "id": "28bbde5a",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Create evenly spaced bars on the marimba (should be taken from urdf in the future)\n",
+    "def generate_marimba_model(num_bars, key_length, key_width):\n",
+    "    # Create evenly spaced keys using NumPy's linspace function\n",
+    "    keys = np.linspace(0, key_width * num_bars, num=num_bars)\n",
+    "\n",
+    "    # Create lines based on the key points \n",
+    "    lines = np.zeros((num_bars, 2, 3))\n",
+    "    lines[:, :, 0] = keys[:, None]\n",
```

Comment:
I was confused by the None in this line.
After some googling I understand it adds a new dimension, but maybe I would suggest using numpy.newaxis here as it makes the operation more readable. https://numpy.org/doc/stable/user/basics.indexing.html#dimensional-indexing-tools
Maybe this is nit picking though

Reply:
I think it is preference, but I like the broadcasting notation with `None`, but I can see your point.